provider/cloudflare/cloudflare.go

```diff
@@ -722,22 +748,34 @@ func (p *CloudFlareProvider) newCloudFlareChange(action changeAction, ep *endpoi
 		comment = p.DNSRecordsConfig.trimAndValidateComment(ep.DNSName, comment, p.ZoneHasPaidPlan)
 	}
 
+	priority := (*uint16)(nil)
+	if ep.RecordType == "MX" {
+		mxRecord, err := endpoint.NewMXTarget(target)
+		if err != nil {
```

Comment:
not covered with unit tests
![Screenshot 2025-06-22 at 10 09 10](https://github.com/user-attachments/assets/c4822d5e-f56e-495d-85fc-7ef89d469473)